### PR TITLE
WT-8501 Create functions to refactor endianness tasks in evergreen.yml

### PR DESCRIPTION
**Pull request by etienneptl**
- New function `fetch artifacts from endian` to fetch little and big endian artifacts
- New function `verify wt datafiles` that executes the `verify_wt_datafiles.sh` script

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -38,29 +38,17 @@ functions:
...
     - command: s3.get
       params:
         aws_key: ${aws_key}
         aws_secret: ${aws_secret}
-        remote_file: wiredtiger/little-endian/${revision}/artifacts/WT_TEST.tgz
+        remote_file: wiredtiger/${endian_format}/${revision}/artifacts/WT_TEST.tgz
         bucket: build_external
-        local_file: WT_TEST-little-endian.tgz
+        local_file: WT_TEST-${endian_format}.tgz
     - command: archive.targz_extract
```
Comment (Siddhartha8899) [inline test/evergreen.yml:49]: I was wondering if we could use `extract_to` here? Like [this](https://github.com/wiredtiger/wiredtiger/pull/7278/files#diff-42f20e21217bd7d4b14489056e1ea24ea2b87599c85413d49182a3b0b721c132L40)

File: test/evergreen.yml [hunk 2]
```diff
@@ -38,29 +38,17 @@ functions:
       remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${dependent_task|compile}_${build_id}.tgz
       bucket: build_external
       extract_to: wiredtiger
-  "fetch artifacts from little-endian" :
+  "fetch artifacts from endian" :
```
Comment (lukech) [inline test/evergreen.yml:41]: "endian" --> "endian format"

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml:49]: My recollection is the `extract_to` setting has a built-in timer that could expire as the artifacts extraction could take a long time (esp on a big-endianness/zSeries machine). Feel free to give it a try with `extract_to` though.

Comment (Siddhartha8899) [reply test/evergreen.yml:49]: I believe @etienneptl  ran a patch build.

Comment (etienneptl) [reply test/evergreen.yml:49]: I am still trying to use the `extract_to` function but no luck so far. I don't know if the Evergreen command cannot archive multiple folders at once or if I am not using it correctly. Will keep you updated.
As a side note, all patch builds are posted in the Jira ticket.

File: test/evergreen.yml (hunk 2 above)
Comment (etienneptl) [reply test/evergreen.yml]: Renamed in 070e926bed58849668f694396f32164b06700ba8

File: test/evergreen.yml (hunk 1 above)
Comment (etienneptl) [reply test/evergreen.yml]: I have added the `extract_to` command in 7b380b5234b79a9ad776492d7755dbac26b072bf.

File: test/evergreen.yml [hunk 3]
```diff
@@ -38,30 +38,33 @@ functions:
...
-        remote_file: wiredtiger/little-endian/${revision}/artifacts/WT_TEST.tgz
+        remote_file: wiredtiger/${endian_format}/${revision}/artifacts/WT_TEST.tgz
         bucket: build_external
-        local_file: WT_TEST-little-endian.tgz
-    - command: archive.targz_extract
-      params:
-        path: "WT_TEST-little-endian.tgz"
-        destination: "wiredtiger/build_posix/test/format"
-  "fetch artifacts from big-endian" :
+        extract_to: wiredtiger/build_posix/test/format
```
Comment (lukech) [inline test/evergreen.yml:48]: It looks like after switching to using `extract_to` we lose the capacity to set the tarball name (or `path` parameter) as in `archive.targz_extract`. Having the file name reflecting the endian format is useful, as we have tests to verify data files from both the same or a different endianness system.

File: test/evergreen.yml [hunk 4]
```diff
@@ -1870,33 +1881,25 @@ tasks:
...
+      - func: "upload endian format artifact"
+        vars:
+          endian_format: big-endian
           local_file: wiredtiger/build_posix/test/format/WT_TEST.tgz
-          bucket: build_external
-          permissions: public-read
-          content_type: application/tar
-          display_name: WT_TEST
-          remote_file: wiredtiger/big-endian/${revision}/artifacts/WT_TEST.tgz
+          remote_file: WT_TEST.tgz
```
Comment (lukech) [inline test/evergreen.yml:1891]: Can we reflect the endianness format in the file name e.g. `WT_TEST-${endian_format}.tgz`?

File: test/evergreen.yml (hunk 3 above)
Comment (lukech) [reply test/evergreen.yml:48]: I made a suggestion below on line 1891.

File: test/evergreen.yml [hunk 5]
```diff
@@ -38,30 +38,33 @@ functions:
...
+        remote_file: wiredtiger/${endian_format}/${revision}/artifacts/WT_TEST.tgz
         bucket: build_external
-        local_file: WT_TEST-little-endian.tgz
-    - command: archive.targz_extract
-      params:
-        path: "WT_TEST-little-endian.tgz"
-        destination: "wiredtiger/build_posix/test/format"
-  "fetch artifacts from big-endian" :
+        extract_to: wiredtiger/build_posix/test/format
+  "fetch artifacts from many-coll" :
```
Comment (etienneptl) [inline test/evergreen.yml:49]: It seems that the merge with `wt-8498-evergreen-improvements` was not done correctly.

Comment (etienneptl) [reply test/evergreen.yml]: Fixed in e2d5c91

File: test/evergreen.yml (hunk 3 above)
Comment (lukech) [reply test/evergreen.yml:48]: Yes. It would help with debugging with a spawn host.

Comment (etienneptl) [reply test/evergreen.yml:48]: I did it in 5141f0800a9fabe9b0ee7c867a0969aff5d57242

File: test/evergreen.yml (hunk 4 above)
Comment (etienneptl) [reply test/evergreen.yml]: See 5141f0800a9fabe9b0ee7c867a0969aff5d57242

File: test/evergreen.yml (hunk 3 above)
Comment (lukech) [reply test/evergreen.yml:48]: Thanks.

File: test/evergreen.yml [hunk 6]
```diff
@@ -38,30 +38,14 @@ functions:
...
       bucket: build_external
       extract_to: wiredtiger
-  "fetch artifacts from little-endian" :
+  "fetch artifacts from endian format" :
     - command: s3.get
       params:
         aws_key: ${aws_key}
         aws_secret: ${aws_secret}
-        remote_file: wiredtiger/little-endian/${revision}/artifacts/WT_TEST.tgz
+        remote_file: wiredtiger/${endian_format}/${revision}/artifacts/${remote_file}.tgz
```
Comment (lukech) [inline test/evergreen.yml:46]: Are you meant to use `${local_file}.tgz` here?

Comment (lukech) [reply test/evergreen.yml:46]: Please ignore this question. I understand how it works now

**Review by lukech — APPROVED** [review]
LGTM

**Review by Siddhartha8899 — APPROVED** [review]

**Pull request merged**